src/Altinn.Platform/Altinn.Platform.Storage/Storage/Controllers/InstancesController.cs

```diff
@@ -320,19 +340,184 @@ public async Task<ActionResult> Post(string appId, int? instanceOwnerId, [FromBo
+        /// </summary>
+        /// <param name="request">The HttpRequest</param>
+        /// <param name="appInfo">The application information</param>
+        /// <returns>The instance object</returns>
+        private async Task<Instance> ReadInstanceTemplateFromBody(HttpRequest request, Application appInfo)
+        {
+            Instance instanceTemplate = null;
+
+            if (MultipartRequestHelper.IsMultipartContentType(request.ContentType))
+            {
+                // Only read the first section of the mulitpart message.
+                MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(request.ContentType);
+                string boundary = MultipartRequestHelper.GetBoundary(mediaType, _defaultFormOptions.MultipartBoundaryLengthLimit);
+
+                MultipartReader reader = new MultipartReader(boundary, request.Body);
+                MultipartSection section = reader.ReadNextSectionAsync().Result;
+
+                bool hasContentDispositionHeader =
+                        ContentDispositionHeaderValue.
+                        TryParse(section.ContentDisposition, out ContentDispositionHeaderValue contentDisposition);
+
+                if (hasContentDispositionHeader)
+                {
+                    // Check if the content disposition name is "instance".
+                    if (contentDisposition.Name.Value == "instance")
+                    {
+                        string contentType = section.ContentType.Split(";")[0];
+
+                        // Check if the content type of the multipart section is declared for the application (for example "application/json").
+                        if (!appInfo.ElementTypes.Exists(e => e.AllowedContentType.Contains(contentType)))
+                        {
+                            instanceTemplate = JsonConvert.DeserializeObject<Instance>(await section.ReadAsStringAsync());
+                            request.Body.Position = 0;
+                            return instanceTemplate;
+                        }
+                    }
+                }
+            }
+            else
+            {
+                // Check if the content type of the multipart section is declared for the application (for example "application/json").
+                if (appInfo.ElementTypes.Exists(e => e.AllowedContentType.Contains(request.ContentType)))
+                {
+                    instanceTemplate = JsonConvert.DeserializeObject<Instance>(await ReadBodyAsync(request));
+                    return instanceTemplate;
+                }
+            }
+
+            return instanceTemplate;
         }
 
-        private string GetUser(ClaimsPrincipal user)
+        /// <summary>
+        /// Reads the body element of HttpRequest as string
+        /// </summary>
+        /// <returns></returns>
+        private async Task<string> ReadBodyAsync(HttpRequest req)
+        {
+            StreamReader streamReader = new StreamReader(req.Body);
```

Comment:
Here you should use the encoding of the contentType!!! If no encoding is present use Encoding.UTF8 as defalt constructor of the StreamReader.

Reply:
Fixed with using Encoding.UTF8 as agreed, with no further checks of incoming encoding.